elasticsearch/src/main/java/org/hibernate/search/elasticsearch/logging/impl/Log.java

```diff
@@ -37,7 +38,7 @@
  * @author Gunnar Morling
  */
 @MessageLogger(projectCode = "HSEARCH")
-public interface Log extends org.hibernate.search.util.logging.impl.Log {
+public interface Log extends BaseHibernateSearchLogger {
```

Comment:
I know it's not _exactly_ the point of this PR, but since it's related... shouldn't we also make sure the "engine" interface isn't used in other modules? And that they rather use the `BaseHibernateSearchLogger` methods on their own child interface?

It seems we currently do use the "engine" logger in other modules, for instance there:

 * `org.hibernate.search.backend.impl.jms.AbstractJMSHibernateSearchController`
 * `org.hibernate.search.elasticsearch.analyzer.impl.NamedElasticsearchAnalyzerReference`
 * ...

Reply:
yes that's the ideal perfect world scenario. 
I'm worried that we're adding a bit too much "red tape" which really doesn't add much value but makes development slower.

For now I'm inclined to let us use a bit more flexibility and choose to share an error, we can refine this after seeing how it works out.

Reply:
updated this comment as it was worded confusing.